criu/arch/x86/cpu.c

```diff
@@ -261,25 +262,37 @@ static int cpu_validate_features(compel_cpuinfo_t *cpu_info)
 #undef __mismatch_fpu_bit
 
 		/*
-		 * Make sure the xsave features are compatible. We already hit the
-		 * issue with libc where we've checkpointed the container on old
-		 * machine but restored on more modern one and libc fetched new
-		 * xsave frame size directly by xsave instruction with greedy
-		 * feature mask causing programs to misbehave.
+		 * Make sure the xsave features are compatible. We were relying
+		 * on our guess about hardware "backward compatibility" and
+		 * used ">" check here for a long time. But it looks better to
+		 * explicitly check that all xfeature bits from source are
+		 * present on the destination.
```

Comment:
I'm not sure anybody would understand ">" here in 2021..
Maybe instead:
```
		 * Make sure the xsave features are compatible. Check that on destination
		 * there are all the features which were on the source.
```
Hmm?

Reply:
actually, "on a destination" and "on a source" (per my limited knowledge of ingles)

Reply:
Makes sense, but for me it sounds better to use definite article here, because we only have one destination and one source and they seam defined for me.